pep/app/pods/components/document/text/component.ts

```diff
@@ -457,14 +462,50 @@ export default class DocumentText extends Component<DocumentTextArgs> {
      * @param {HTMLElement} element
      * @memberof DocumentText
      */
+    /**
+     *
+     *
+     * @param {HTMLElement} element
+     * @memberof DocumentText
+     */
     @action
-    setupListeners(element: HTMLElement) {
+    async setupListeners(element: HTMLElement) {
         this.containerElement = element;
         this.scrollableElement = this.containerElement?.closest('.page-content-inner');
         scheduleOnce('afterRender', this, this.attachTooltips);
         if (this.args.page) {
-            this.scrollToPageOrTarget(this.args.page);
+            await this.scrollToPageOrTarget(this.args.page);
         }
+        Object.values(this.containerElement.querySelectorAll('div.pagebreak')).forEach((item) => {
+            const observer = new IntersectionObserver(
```

Comment:
@bakerac4 not that it'll fix your query param issue, but I just noticed this. Looking the `IntersectionObserver` I _think_ you should be able to create just one instance of it, and in your loop here you only call `observer.observe()` for each element. That way you don't have like hundreds of `IntersectionObserver`'s on really long documents.